### PR TITLE
[2d] variable assigned a value which is never used

### PR DESCRIPTION
**Pull request by divmadan**
- Fixes a part of #3609
- Solves the issue for files in 2d directory (only)

### CODE REVIEW

**Comment by SunBlack** [comment]
As far as I can see in `hessianBlob`:
* Scope of `local_max` could reduced to `const float local_max = cornerness[k][i][j];`
*  Value of `scale_max` is not used anywhere, so it can be reduced and `if (cornerness[k][i][j] > scale_max)` can be changed to `if (cornerness[k][i][j] > std::numeric_limits<float>::min())`

**Comment by SunBlack** [comment]
Can it be that `hessianBlob` is broken?
Shouldn't this
https://github.com/PointCloudLibrary/pcl/blob/9541a6ff01562fc34eaec23f8fe95c8934b6e826/2d/include/pcl/2d/impl/keypoint.hpp#L233
be `output[i][j]` instead of `output[i][i]`

File: 2d/include/pcl/2d/impl/kernel.hpp [hunk 1]
```diff
@@ -131,9 +130,10 @@ pcl::kernel<PointT>::loGKernel(pcl::PointCloud<PointT>& kernel)
 
   for (int i = 0; i < kernel_size_; i++) {
     for (int j = 0; j < kernel_size_; j++) {
+      float temp = 0;
       int iks = (i - kernel_size_ / 2);
       int jks = (j - kernel_size_ / 2);
-      temp = float(double(iks * iks + jks * jks) / sigma_sqr);
+      float temp = float(double(iks * iks + jks * jks) / sigma_sqr);
```
Comment (kunaltyagi) [inline 2d/include/pcl/2d/impl/kernel.hpp:136]: Why did you add L133?

Comment (divmadan) [reply 2d/include/pcl/2d/impl/kernel.hpp:135]: My mistake. The [Suggested change](https://github.com/PointCloudLibrary/pcl/pull/3857#discussion_r403250720) didn't remove the line, and I missed that.

**Comment by divmadan** [comment]
@SunBlack 
> * Scope of `local_max` could reduced to `const float local_max = cornerness[k][i][j];`

Agree.
> * Value of `scale_max` is not used anywhere, so it can be reduced and `if (cornerness[k][i][j] > scale_max)` can be changed to `if (cornerness[k][i][j] > std::numeric_limits<float>::min())`

Why should we check every time with `std::numeric_limits<float>::min()`?
It stores the previous value of `cornerness[k][i][j]` for comparison with the updated value after the loop rerun.  It's value is not reset to `std::numeric_limits<float>::min()` in the loop, that initialisation is outside the loop.

Correct me if I'm wrong.

**Comment by SunBlack** [comment]
You are right, I missed the loop `for (int k = 0; k < num_scales; k++) {` nevertheless: The scope of `scale_max` can be reduced ;-)

**Comment by taketwo** [comment]
:man_facepalming: should have squashed

**Comment by SunBlack** [comment]
Force push to master :-P

**Comment by taketwo** [comment]
I realized I made a mistake the same moment I pressed "merge", so I guess it would have been kinda okayish to force push immediately after... But our `master` branch is protected anyway :man_shrugging:

**Comment by SunBlack** [comment]
Isn't it just a setting, which you can turn off shortly?
